cont3xt/vueapp/src/utils/HighlightableText.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <span>
+    <template v-if="highlights != null">
+      <span
+          v-for="{start, end, highlighted} in spans"
+          :class="{highlight: highlighted}"
+          :key="`span-${start}-${end}`">{{text.substring(start, end)}}</span>
+    </template>
+    <template v-else>{{content}}</template>
+  </span>
+</template>
+
+<script>
+export default {
+  name: 'HighlightableText',
+  props: {
+    content: { // content can be any type with a valid .toString() -- or null/undefined
+      validator (prop) {
+        if (prop == null ||
+            typeof prop === 'string' ||
+            typeof prop === 'number' ||
+            typeof prop === 'boolean' ||
+            typeof prop === 'object') return true;
+        console.log(`WARNING - unsupported type ${typeof prop} in highlight-text`);
+        return false;
+      },
+      required: true
+    },
+    // an array of spans (ordered by start, non-overlapping), ex. [{start: 0, end: 2}, {start: 3, end: 7}]
+    highlights: {
+      type: Array,
+      default () {
+        return null;
+      }
+    }
+  },
+  computed: {
+    text () {
+      return this.content ? this.content.toString() : undefined;
+    },
+    spans () {
+      if (this.highlights == null) {
+        return [{ start: 0, end: this.text.length, highlighted: false }];
+      }
+
+      const spanList = [];
+      const expandedHighlights = [{ start: -1, end: 0 }, ...this.highlights];
+
+      const correctHighlightStart = (start) => {
+        return Math.min(start, this.text.length - 3);
+      };
+
+      for (const [i, highlightedSpan] of expandedHighlights.entries()) {
```

Comment:
Personally I hate entries :) Just curious, is there an advantage over a normal for loop?

Reply:
nope, just wondering